File: src/main/java/com/intetics/lukyanenko/dao/OrderDetailDAO.java
package com.intetics.lukyanenko.dao;

import com.intetics.lukyanenko.models.Order;
import com.intetics.lukyanenko.models.OrderDetail;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;

public interface OrderDetailDAO
        extends CommonDAO<OrderDetail>
{
  OrderDetail findGoodsItemInOrder(Order order, Integer goodsItemId);
  
  List<OrderDetail> getListForOrder(Order order, SubdetailsFillable subDetails);
  
  interface SubdetailsFillable
  {
    void fill(OrderDetail detail, ResultSet resultSet, String fieldNamePrefix) throws SQLException;
  }
}

File: src/main/java/com/intetics/lukyanenko/dao/OrderDetailDAO.kt
package com.intetics.lukyanenko.dao

import com.intetics.lukyanenko.models.Order
import com.intetics.lukyanenko.models.OrderDetail

interface OrderDetailDAO : CommonDAO<OrderDetail> {
    fun findGoodsItemInOrder(order: Order, goodsItemId: Int): OrderDetail?

    fun getListForOrder(order: Order): MutableList<OrderDetail>
}
